tests/export/image_resize/test_image_resize.py

```diff
@@ -0,0 +1,175 @@
+        image_data = get_img('image1.png', as_binary=True)
+
+        ir = ImageResizer(image_data, 'image1.png', '0 px', '10 px')
+        self.assertFalse(ir.has_height_and_width())
+
+        ir = ImageResizer(image_data, 'image1.png', '10 px', '0 px')
+        self.assertFalse(ir.has_height_and_width())
+
+        ir = ImageResizer(image_data, 'image1.png', '0 px', '0 px')
+        self.assertFalse(ir.has_height_and_width())
+
+    def test_init_image(self):
+        image_data = get_img('image1.png', as_binary=True)
+        ir = ImageResizer(image_data, 'image1.png', '100 px', '100 px')
+        ir.init_image()
+        self.assertEqual(ir.image, Image.open(StringIO(image_data)))
+
+        image_data = get_img('image1.data', as_binary=True)
+        ir = ImageResizer(image_data, 'image1.png', '100 px', '100 px')
+        ir.init_image()
+        # here the final Image object should be the same as we create it from binary data
+        self.assertEqual(ir.image, Image.open(StringIO(get_img('image1.png', as_binary=True))))
+
+    def test_init_image_exception(self):
+        ir = ImageResizer(b'test_data3434', 'image1.png', '100 px', '100 px')
+
+        self.assertRaises(IOError, ir.init_image)
+
+    def test_resize_image_skip(self):
+        image_data = get_img('image1.data', as_binary=True)
+
+        ir = ImageResizer(image_data, 'image1.png', '48 px', '48 px')
+        ir.init_image()
+
+        result = ir.resize_image()
+
+        self.assertIsNone(result)
+
+    def test_resize_image_success(self):
+        image_data = get_img('image1.png', as_binary=True)
+
+        ir = ImageResizer(image_data, 'image1.png', '48 px', '48 px')
+
+        ir.init_image()
+        result = ir.resize_image()
+
+        self.assertTrue(result)
+
+        self.assertEqual(ir.image.size, (48, 48))
+
+    def test_resize_image_keep_original(self):
+        image_data = get_img('image1.png', as_binary=True)
+
+        ir = ImageResizer(image_data, 'image1.png', '100 px', '100 px')
+
+        ir.init_image()
+
+        result = ir.resize_image()
+
+        self.assertFalse(result)
+
+        self.assertEqual(ir.image.size, (100, 100))
+
+    def test_resize_image_change_to_gif(self):
+        image_data = get_img('image2.tif', as_binary=True)
+
+        ir = ImageResizer(image_data, 'image2.tif', '50 px', '50 px')
+
+        ir.init_image()
+        result = ir.resize_image()
+
+        self.assertTrue(result)
+
+        self.assertEqual(ir.image_format, 'GIF')
+
+        image = Image.open(StringIO(image_data)).resize((50, 50), Image.ANTIALIAS)
+
+        self.assertEqual(image, ir.image)
+        self.assertEqual('image2.tif', ir.filename)
+
+    def test_update_filename(self):
+        image_data = get_img('image1.png', as_binary=True)
+
+        ir = ImageResizer(image_data, 'image1.png', '48 px', '48 px')
+
+        ir.init_image()
+
+        ir.update_filename()
+
+        self.assertEqual(ir.filename, 'image1.png')
```

Comment:
The filename in this test is the same because it's a PNG format, right?

If we pass in a `.tiff` file, would the resulting filename be a `.gif`?